### PR TITLE
[INJI-175] Custom Secure keystore integration and show warning message when not supported

### PR DESCRIPTION
**Pull request by dhivya0413**
Android which has hardware keystore uses custom Secure-keystore SDK(https://mosip.atlassian.net/browse/INJI-214), otherwise it uses default react-native-keychain library. 

- Flow modified to encrypt, decrypt, hmac generation, keypair generation(for esignet) handles by custom library
- Show warning message if hardware keystore doesn't supported only for first time
- Locales added for the warning message

https://mosip.atlassian.net/browse/INJI-175

Note - library version to be modified once we get repo access

### CODE REVIEW

File: locales/ara.json [hunk 1]
```diff
@@ -253,6 +253,11 @@
       "vcIsTampered":{
         "title": "تم العبث ببيانات VC",
         "message": "تتم إزالة VCs المتأثر من التطبيق"
+      },
+      "keystoreNotExists": {
+        "title": "لم يتم العثور على Secure Key Storage",
```
Comment (vijay151096) [inline locales/ara.json:258]: can we translate "Secure Key Storage" literal in native language

File: package.json [hunk 2]
```diff
@@ -76,6 +76,7 @@
     "react-native-safe-area-context": "3.3.2",
     "react-native-screens": "~3.10.1",
     "react-native-secure-key-store": "^2.0.10",
+    "react-native-secure-keystore": "file:.yalc/react-native-secure-keystore",
```
Comment (vijay151096) [inline package.json:79]: can we remove the yalc and using the commit sha or tags to access the library

File: machines/store.ts [hunk 3]
```diff
@@ -114,10 +135,17 @@ export const storeMachine =
...
-              target: 'resettingStorage',
-            },
+            KEY_RECEIVED: [
+              {
+                cond: 'isCustomSecureKeystore',
+                target: 'ready',
+              },
+              {
+                actions: 'setEncryptionKey',
+                target: 'resettingStorage',
```
Comment (vharsh) [inline machines/store.ts:145]: I might be wrong on this one but is the storage getting cleared each time the key is received on a phone which has a hardware keystore? Can this be tested once?

**Review by vharsh — COMMENTED** [review]
minor suggestions & doubts

File: machines/QrLoginMachine.ts [hunk 4]
```diff
@@ -352,9 +355,14 @@ export const qrLoginMachine =
...
-            context.selectedVc.walletBindingResponse?.walletBindingId
-          );
+          let privateKey;
+
+          if (!isCustomSecureKeystore()) {
+            privateKey = await getPrivateKey(
+              context.selectedVc.walletBindingResponse?.walletBindingId
+            );
+          }
+
```
[the remark this answers is not in the document]
Comment (dhivya0413) [reply machines/QrLoginMachine.ts:365]: iOS and android lacks hardware keystore uses private key to get JWT token in order not to break the existing implementation.

File: machines/settings.ts [hunk 5]
```diff
@@ -27,6 +29,7 @@ const model = createModel(
     isBiometricUnlockEnabled: false,
     credentialRegistry: HOST,
     appId: null,
+    hasUserShownWithHardwareKeystoreNotExists: false,
```
[the remark this answers is not in the document]
Comment (dhivya0413) [reply machines/settings.ts:32]: this variable is not defines about hardwarekeystore exists or not, rather it's about already being shown to user.

File: shared/cryptoutil/cryptoUtil.ts [hunk 6]
```diff
@@ -11,21 +13,18 @@ export async function getJwt(
...
 
-    const key = forge.pki.privateKeyFromPem(privateKey);
-    const md = forge.md.sha256.create();
     const header = {
       'alg': 'RS256',
       //'kid': keyId,
       'x5t#S256': thumbprint,
     };
-    var myJSON =
+    var payloadJSON =
```
[the remark this answers is not in the document]
Comment (dhivya0413) [reply shared/cryptoutil/cryptoUtil.ts:27]: addressed

File: shared/cryptoutil/cryptoUtil.ts [hunk 7]
```diff
@@ -11,21 +13,18 @@ export async function getJwt(
...
       '{"iss": "' +
       config.issuer +
```
[the remark this answers is not in the document]
Comment (dhivya0413) [reply shared/cryptoutil/cryptoUtil.ts:29]: addressed

**Pull request closed without merging**